codegen/settingsgen.py

```diff
@@ -363,7 +363,8 @@ def _populate_classes(parent_dir):
                     stubf.write(f"{istr1}child_names = ...\n")
 
                 for child in child_names:
-                    child_class_strings.append(f"{child}={child}_cls")
+                    child_cls = cls._child_classes[child]
+                    child_class_strings.append(f"{child}={child_cls.__name__}_cls")
```

Comment:
The above code will populate the actual class name (when there is a clash, the class name will be child name with an integer suffix). We are already doing the same for arguments, but should be done for all child types.